scrapi/harvesters/dataone.py

```diff
@@ -124,35 +126,32 @@ class DataOneHarvester(XMLHarvester):
             ('formatId', ("str[@name='formatId']/node()")),
             ('formatType', ("str[@name='formatType']/node()")),
             ('identifier', ("str[@name='identifier']/node()")),
-            ('investigator', "arr[@name='investigator']/str/node()"),
-            ('origin', "arr[@name='origin']/str/node()"),
-            ('isPublic', ("bool[@name='isPublic']/node()")),
             ('readPermission', "arr[@name='readPermission']/str/node()"),
             ('replicaMN', "arr[@name='replicaMN']/str/node()"),
             ('replicaVerifiedDate', "arr[@name='replicaVerifiedDate']/date/node()"),
             ('replicationAllowed', ("bool[@name='replicationAllowed']/node()")),
             ('numberReplicas', ("int[@name='numberReplicas']/node()")),
             ('preferredReplicationMN', "arr[@name='preferredReplicationMN']/str/node()"),
-            ('resourceMap', "arr[@name='resourceMap']/str/node()"),
             ('rightsHolder', ("str[@name='rightsHolder']/node()")),
             ('scientificName', "arr[@name='scientificName']/str/node()"),
             ('site', "arr[@name='site']/str/node()"),
             ('size', ("long[@name='size']/node()")),
-            ('sku', ("str[@name='sku']/node()")),
             ('isDocumentedBy', "arr[@name='isDocumentedBy']/str/node()"),
             ('serviceID', "str[@name='id']/node()")
         ),
         'freeToRead': {
             'startDate': ("bool[@name='isPublic']/node()", "date[@name='dateModified']/node()", lambda x, y: parse(y[0]).date().isoformat() if x else None)
         },
-        'contributors': ("str[@name='author']/node()", "str[@name='submitter']/node()", "arr[@name='origin']/str/node()", process_contributors),
+        'contributors': ("str[@name='author']/node()", "str[@name='submitter']/node()", "arr[@name='origin']/str/node()", "arr[@name='investigator']/str/node()", process_contributors),
         'uris': {
             'canonicalUri': ("str[@name='id']/node()", "//str[@name='dataUrl']/node()", lambda x, y: y[0] if 'http' in single_result(y) else x[0] if 'http' in single_result(x) else ''),
+            'objectUri': "str[@name='resourceMap']/node()".replace("doi:", "http://dx.doi.org/")
```

Comment:
The format for scrAPI schemas is a little weird, but the first element here is an xpath command (```str[@name='resourceMap']/node()```). scrAPI will run this xpath command and return the results. However, if there is some post-processing that you want to do, you can turn the value into a tuple (```(elem1, elem2)```), and make the last element of the tuple a function. scrAPI will take the return values of all of the previous xpath commands (or functions, but usually xpath commads), and feed them into the last function as arguments.

So I think what you were trying to accomplish is achieved like this:
```python
...
{
...
    'objectUri': "str[@name='resourceMap']/node()", lambda x: x.replace('doi:', 'http://dx.doi.org/')
}
...
```

In case you haven't seen them before, lambdas are anonymous functions
```python
add1 = lambda x: x + 1
```
is equivalent to
```
def add1(x):
    return x + 1
```

They are very useful in situations like this, because we have many oneoff functions that are very simple, and we don't really need them to clutter the namespace.